### PR TITLE
sim_vehicle.py: add arguments to pass args to waf build and configure

### LINKED ISSUE #4749: sim_vehicle.py - be able to pass flags to waf configure and build steps
We need a way to pass options through to waf from sim_vehicle.py?  I need to pass the "--check-cxx-compiler g++" option to the waf configure step but not the build step.  We already have --mavproxy-args and --sitl-instance-args, so could we have --configure-args and --build-args ?

Peter has volunteered :-)

Thanks, Grant.

### PR DESCRIPTION
**Pull request by peterbarker**
Will close #4749

### CODE REVIEW

File: Tools/autotest/sim_vehicle.py [hunk 1]
```diff
@@ -239,6 +239,8 @@ def find_root_dir():
 group_build.add_option("-b", "--build-target", default=None, type='string', help="override SITL build target")
 group_build.add_option("-s", "--build-system", default="waf", type='choice', choices=["make", "waf"], help="build system to use")
 group_build.add_option("", "--no-rebuild-on-failure", dest="rebuild_on_failure", action='store_false', default=True, help="if build fails, do not clean and rebuild")
+group_build.add_option("", "--waf-configure-arg", action="append", dest="waf_configure_args", type="string", default=[], help="extra arguments to pass to waf in its configure step")
+group_build.add_option("", "--waf-build-arg", action="append", dest="waf_build_args", type="string", default=[], help="extra arguments to pass to waf in its build step")
```
Comment (guludo) [inline Tools/autotest/sim_vehicle.py]: I think those lines are getting too long.

**Comment by gmorph** [comment]
I've tested passing an option to the configure step of waf using --waf-configure-arg and that works so 
LGTM
Thanks Peter!

**Comment by peterbarker** [comment]
Thanks @gmorph.

Pushed.

File: Tools/autotest/sim_vehicle.py (hunk 1 above)
Comment (peterbarker) [inline Tools/autotest/sim_vehicle.py]: @guludo I'll see if I can neaten them next time I've got an excuse to code-churn there.  Possibly just breaking "help" onto the next line will be sufficient.

I tend to maximise my window when I work on chunks of code like that, so it doesn't really bother me.

**Pull request closed without merging**